.github/workflows/release-snapshot.yaml

```diff
@@ -0,0 +1,34 @@
+name: Release Snapshot
+
+on:
+  workflow_dispatch:
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+env:
+  SUI_VERSION: mainnet-v1.25.3
+
+jobs:
+  release-snapshot:
+    name: Release Snapshot
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Setup Sui CLI and install dependencies
+        uses: ./.github/actions/install
+        with:
+          SUI_VERSION: ${{ env.SUI_VERSION }}
+
+      - name: Build TS and SUI modules
+        run: npm run build
+
+      - name: Set Snapshot Version
+        run: |
+          npm version 0.0.0-snapshot.$(git rev-parse --short HEAD) --git-tag-version=false
+
+      - name: Publish Snapshot to NPM
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > .npmrc
```

Comment:
why not just set the registry during the node setup?

Reply:
have a better approach here https://github.com/axelarnetwork/axelar-cgp-sui/pull/85/commits/ddd1723c6f1bbc22a25c59078c8e27549b2f6c43